Assets/Scripts/FpsController.cs

```diff
@@ -0,0 +1,178 @@
+        _lookPitch = cam.transform.eulerAngles.x;
+    }
+
+    void Update()
+    {
+        Look();
+        MovementInput();
+    }
+
+    void FixedUpdate()
+    {
+        // Move
+        Vector3 moveVector = GetMovementVector();
+        ApplyDirectionalSpeedMultipliers(ref moveVector);
+        Move(moveVector * Time.fixedDeltaTime);
+
+        JumpLogic();
+
+        UpdateGroundedState();
+    }
+
+    void ApplyDirectionalSpeedMultipliers(ref Vector3 vector)
+    {
+        if (_inputVector.y < 0.0f)
+        {
+            vector = vector.normalized * backSpeed;
+        }
+        else
+        {
+            vector = vector.normalized * forwardSpeed;
+        }
+    }
+
+    Vector3 GetMovementVector()
+    {
+        return transform.TransformVector(new Vector3(_inputVector.x, 0.0f, _inputVector.y));
+    }
+
+    void Move(Vector3 displacement)
+    {
+        transform.position += displacement;
+    }
+
+    void Look()
+    {
+        float mouseDeltaYaw = Input.GetAxisRaw("Mouse X");
+        float mouseDeltaPitch = Input.GetAxisRaw("Mouse Y");
+
+        _lookYaw += mouseDeltaYaw;
+        _lookYaw = WrapAroundAngleDegrees(_lookYaw);
+        transform.rotation = Quaternion.Euler(transform.eulerAngles.x, _lookYaw, transform.eulerAngles.z);
+
+        _lookPitch -= mouseDeltaPitch;
+        _lookPitch = ClampPitch(_lookPitch);
+        cam.transform.rotation = Quaternion.Euler(_lookPitch, cam.transform.eulerAngles.y, cam.transform.eulerAngles.z);
+    }
+
+    void MovementInput()
+    {
+        _inputVector = new Vector2(Input.GetAxis("Horizontal"), Input.GetAxis("Vertical"));
+
+        if (_jumpTimeSinceInput >= jumpInputBufferTime)
+        {
+            _jumpInput = Input.GetButtonDown("Jump");
+            if (_jumpInput)
+            {
+                _jumpTimeSinceInput = 0.0f;
+            }
+        }
+        else
+        {
+            _jumpTimeSinceInput += Time.deltaTime;
+        }
+    }
+
+    void UpdateGroundedState()
+    {
+        if (Physics.Raycast(transform.position, Vector3.down, out RaycastHit hitInfo)
+        && body.velocity.y < 0.01f
+        && hitInfo.distance <= 2.0f / 2.0f)
+        {
+            _grounded = true;
+        }
+        else
+        {
+            _grounded = false;
+        }
+    }
+
+    void JumpLogic()
+    {
+        if (_jumpInput && _grounded)
+        {
+            _jumpInput = false;
+            _jumpTimeSinceInput = jumpInputBufferTime;
+
+            Vector3 keepVelocities = Vector3.right + Vector3.forward;
+            body.velocity = Vector3.Scale(body.velocity, keepVelocities);
+            body.AddForce(Vector3.up * jumpForce, ForceMode.Impulse);
+        }
+    }
+
+    float WrapAroundAngleDegrees(float angle)
+    {
+        if (angle > 360.0f)
+        {
+            return angle - 360.0f;
+        }
+        else if (angle < -360.0f)
+        {
+            return angle + 360.0f;
+        }
+
+        return angle;
```

Comment:
nit: Suggest using modulo operator here instead.

For example

```suggestion
        return (angle + 360) % 360;
```